src/client/auth.ts

```diff
@@ -87,12 +87,26 @@ export async function auth(
   { serverUrl,
     authorizationCode,
     scope,
+    resourceMetadataUrl
   }: {
     serverUrl: string | URL;
     authorizationCode?: string;
     scope?: string;
-  }): Promise<AuthResult> {
-  const metadata = await discoverOAuthMetadata(serverUrl);
+    resourceMetadataUrl?: URL }): Promise<AuthResult> {
+
+  let authorizationServerUrl = serverUrl;
+  try {
+    const resourceMetadata = await discoverOAuthProtectedResourceMetadata(
+      resourceMetadataUrl || serverUrl);
+
+    if (resourceMetadata.authorization_servers && resourceMetadata.authorization_servers.length > 0) {
```

Comment:
It feels like we could possibly expose a better interface in this scenario - perhaps a function on the provider that would be called when there is more than one authorization server returned from the protected resource metadata that would pass in the array of servers and utilize the one returned from the function. Happy to PR this in if you feel like it makes sense!